src/common/query/src/error.rs

```diff
@@ -169,6 +169,21 @@ pub enum Error {
         source: BoxedError,
     },
 
+    #[snafu(display("Failed to join thread"))]
+    ThreadJoin { location: Location },
```

Comment:
Is there a source error?

Reply:
Already logged it. It can be present in the `Error` because it's tagged as `Any`.